File: MovieTVShowWatchlist/src/MovieTVShowWatchlist.Core/Model/FavoriteAggregate/Favorite.cs
namespace MovieTVShowWatchlist.Core;

public class Favorite
{
    public Guid FavoriteId { get; set; }
    public Guid UserId { get; set; }
    public Guid ContentId { get; set; }
    public ContentType ContentType { get; set; }
    public DateTime AddedDate { get; set; }
    public string? FavoriteCategory { get; set; }
    public int RewatchCount { get; set; }
    public string? EmotionalSignificance { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public User User { get; set; } = null!;
}

File: MovieTVShowWatchlist/src/MovieTVShowWatchlist.WebApp/projects/movie-tv-show-watchlist/src/app/models/favorite.ts
export interface Favorite {
  favoriteId: string;
  contentId: string;
  contentType: string;
  addedDate: Date;
  favoriteCategory?: string;
  rewatchCount: number;
  emotionalSignificance?: string;
}

export interface CreateFavoriteRequest {
  contentType: string;
  favoriteCategory?: string;
  emotionalSignificance?: string;
}
